d3a_interface/sim_results/bills.py

```diff
@@ -250,13 +254,38 @@ def _accumulate_grid_fee_charged(self, area_dict, area_core_stats):
             self._accumulate_grid_fee_charged(child, area_core_stats)
 
     def update(self, area_dict, area_core_stats):
+        # Updates the self.market_fees dict which keeps track of the accumulated market fees for
+        # each area.
         self._update_market_fees(area_dict, area_core_stats)
+        # Calculates the cumulative_fee_charged_per_market , which in reality is the sum of the
+        # grid fees for all markets for the whole simulation duration.
         self._accumulate_grid_fee_charged(area_dict, area_core_stats)
+        # Generate tree of energy bills, following the area_dict structure. Uses uuids
         bills = self._energy_bills(area_dict, area_core_stats)
         flattened = {}
+        # Flatten the energy bill tree.
+        # The flattened dict contains only area_uuid -> energy_bills_dict, no children
+        # or aggregated data like Totals, External Trades etc.
         self._flatten_energy_bills(bills, flattened)
-        self.bills_results = self._accumulate_by_children(area_dict, flattened, {})
-        self._bills_for_redis(area_dict, deepcopy(self.bills_results))
+        # Adds children to the flattened dict by iterating over the area_dict, finding out the
+        # children of each area, and copying the children bills under their respective parent. Only
+        # 1 level child hierarchy.
+        # The same function populates the Eternal Trades, Accumulated Trades, Market Fees and
+        # Totals bills for the areas that are not leaves (and have children).
+        # The naming format of the areas is still uuid
+        bills = self._accumulate_by_children(area_dict, flattened, {})
+        # Keep the state of the unformatted bills in order to be reused by _energy_bills method.
+        # We need this in order to read past results when calculating the new energy bills.
+        self.current_raw_bills = deepcopy(bills)
+        # Converts the children of the bills from uuids to names, because the UI uses these
+        # as headers in the table where these data are reported.
+        bills = self._swap_children_uuids_to_names(area_dict, bills)
+        # Converts the keys of the result dict from uuids to names, in order to end up in
+        # the CSV files in a human-readable format.
+        self.bills_results = self._bills_local_format(area_dict, bills)
```

Comment:
We could save this effort for the results_service, right?
If you agree, we could add a `should_export_plots` parameter to the `MarketEnergyBills` (just like in other results endpoints). What do you think?

Reply:
Yep, why not. I'll add it, thanks